src/lottery-machine/lottery-machine.service.js

```diff
@@ -0,0 +1,301 @@
+    return lotteryTicketNumber.filter((ticketNumber) =>
+      this.#lotteryMachineModel.getWinningNumbers().includes(ticketNumber),
+    ).length;
+  }
+
+  /**
+   *
+   * @param {Array<number>} lotteryTicketNumber
+   * @returns {boolean}
+   */
+  #hasBonusNumber(lotteryTicketNumber) {
+    return lotteryTicketNumber.includes(this.#lotteryMachineModel.getBonusNumber());
+  }
+
+  /**
+   *
+   * @param {WinningStatistics} winningStatistics
+   * @param {number} matchedLotteryNumber
+   * @param {boolean} hasBonusNumber
+   * @returns {WinningStatistics}
+   */
+  #calculateNextWinningStatistics(winningStatistics, matchedLotteryNumber, hasBonusNumber) {
+    if (matchedLotteryNumber < 3) {
+      return { ...winningStatistics };
+    }
+
+    if (hasBonusNumber && matchedLotteryNumber === 5) {
+      return {
+        ...winningStatistics,
+        bonus: winningStatistics.bonus + 1,
+      };
+    }
+
+    return {
+      ...winningStatistics,
+      [matchedLotteryNumber]: winningStatistics[matchedLotteryNumber] + 1,
+    };
+  }
+
+  /**
+   *
+   * @param {WinningStatistics} winningStatistics
+   * @param {Array<number>} lotteryTicketNumber
+   * @returns {WinningStatistics}
+   */
+  #calculateWinningStatistics(winningStatistics, lotteryTicketNumber) {
+    const matchedLotteryNumber = this.#countMatchingLotteryNumber(lotteryTicketNumber);
+    const hasBonusNumber = this.#hasBonusNumber(lotteryTicketNumber);
+
+    return this.#calculateNextWinningStatistics(
+      winningStatistics,
+      matchedLotteryNumber,
+      hasBonusNumber,
+    );
+  }
+
+  /**
+   *
+   * @param {Array<number[]>} lotteryTicketNumbers
+   * @returns {WinningStatistics}
+   */
+  #createWinningStatistics(lotteryTicketNumbers) {
+    return lotteryTicketNumbers.reduce(
+      (winningStatistics, lotteryTicketNumber) =>
+        this.#calculateWinningStatistics(winningStatistics, lotteryTicketNumber),
+      { ...LotteryMachineService.STRATEGY.WINNING_STATISTICS },
+    );
+  }
+
+  /**
+   *
+   * @returns {{ winningStatistics: WinningStatistics; winningAmount: WinningAmount }}
+   */
+  generateWinningStatistics() {
+    return {
+      winningStatistics: this.#createWinningStatistics(
+        this.#lotteryMachineModel.getLotteryTicketNumbers(),
+      ),
+      winningAmount: LotteryMachineService.STRATEGY.WINNING_AMOUNT,
+    };
+  }
+
+  /**
+   *
+   * @param {string} winningAmount
+   * @returns {number}
+   */
+  #parseWinningAmount(winningAmount) {
+    return Number(winningAmount.split(',').join(''));
+  }
+
+  /**
+   *
+   * @param {WinningStatistics} winningStatistics
+   * @param {WinningAmount} winningAmount
+   * @returns {number}
+   */
+  #calculateTotalWinningAmount(winningStatistics, winningAmount) {
+    return Object.entries(winningStatistics).reduce(
+      (totalWinningAmount, [winningNumber, winningCount]) => {
+        if (winningCount === 0) {
+          return totalWinningAmount;
+        }
```

Comment:
사소한 것이지만 not 연산자를 사용하면 저는 더 깔끔해 보이더라구요!

```suggestion
        if (!winningCount) {
          return totalWinningAmount;
        }
```